### PR TITLE
MCS-1743 console output to log file

### PR DESCRIPTION
**Pull request by jaudick**
[MCS](https://github.com/NextCenturyCorporation/MCS/compare/MCS-1743-console-output-to-log-file?expand=1)

This side of the branch makes a log file with all of the output from the unity console. I also added a more in depth output of what happened each action. The output and updating of the log file only happens if `outputLogFile` is set to true in `MCSMain`

### CODE REVIEW

File: unity/Assets/Scripts/MCSController.cs [hunk 1]
```diff
@@ -533,7 +539,27 @@ private enum HapticFeedback {
         // Never let the placeable objects ignore the physics simulation (they should always be affected by it).
         controlCommand.placeStationary = false;
 
+        string imageCoords = "objectImageCoordsX=" + controlCommand.objectImageCoords.x + ",objectImageCoordsY=" + controlCommand.objectImageCoords.y;
+        string receptacleImageCoords = "receptacleObjectImageCoordsX=" + controlCommand.receptacleObjectImageCoords.x + ",receptacleObjectImageCoordsY=" + controlCommand.receptacleObjectImageCoords.y;
+        string actionParameters = "None";
+        if (this.lastInputAction == InputAction.OTHER) {
```
Comment (ThomasSchellenbergNextCentury) [inline unity/Assets/Scripts/MCSController.cs:545]: What's the "OTHER" action?

File: unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs [hunk 2]
```diff
@@ -2026,13 +2026,6 @@ protected void sopApplyForce(ServerAction action, SimObjPhysics sop, float lengt
                     print("object id: " + feedback.objectId);
                     print("armslength: " + feedback.armsLength);
                     #endif
-                    actionFinished(true, feedback);
-                }
-
-                //why is this here?
-                else
-                {
-                    actionFinished(true);
```
Comment (ThomasSchellenbergNextCentury) [inline unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs:2035]: Just out of curiosity -- what's up with this?

**Comment by ThomasSchellenbergNextCentury** [comment]
@jaudick It feels like this PR has two separate purposes:
1. Log all of the console messages to file, for reference while debugging
2. Log all of the actions taken in the Unity Editor, to replicate them exactly when running via Python

I'd recommend that we save this information in two separate log files. Your current log file serves the purpose for number 1. For purpose number two, we can have another log file where each action and its parameters are saved in an individual line, like this:

```
MoveAhead
MoveAhead
LookDown
PickupObject,objectImageCoordsX=300,objectImageCoordsY=200
```

Then we could use the existing [run_action_file script](https://github.com/NextCenturyCorporation/MCS/blob/development/machine_common_sense/scripts/run_action_file.py) on it. Maybe doing this would also solve the issues you said you're having getting some of the actions to work properly. This also means you wouldn't need a separate python script just for this purpose, which can also be easily broken if someone changes the format of the log messages in the future.

File: unity/Assets/Scripts/MCSController.cs (hunk 1 above)
Comment (jaudick) [reply unity/Assets/Scripts/MCSController.cs:545]: Any action other than Move, Rotate or Pass

File: unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs (hunk 2 above)
Comment (jaudick) [reply unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs:2035]: Somehow I thought this was some unnecessary redundant code but looking back now its just confusing. I can add it back.

**Comment by jaudick** [comment]
@ThomasSchellenbergNextCentury That change to two log files here works for me

**Comment by jaudick** [comment]
@ThomasSchellenbergNextCentury Just pushed a change and closed the python branch

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]
LGTM

**Pull request merged**